lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -159,6 +159,7 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     }
 
     if (killAfterTimeoutThread) {
+      usleep(1000); // Give the thread some time to actually reach wait_for()
```

Comment:
True, this is just making it less likely. What do you think should we do instead, another `std::conditional` that the sigkill thread notifies on?